Framework/Nexus/src/NexusClasses.cpp

```diff
@@ -373,9 +285,10 @@ std::vector<char> &NXBinary::binary() {
  */
 NXRoot::NXRoot(std::string fname) : m_filename(std::move(fname)) {
   // Open NeXus file
-  NXstatus stat = NXopen(m_filename.c_str(), NXACC_READ, &m_fileID);
-  if (stat == NXstatus::NX_ERROR) {
-    std::cout << "NXRoot: Error loading " << m_filename;
+  try {
+    m_fileID = new ::NeXus::File(m_filename, NXACC_READ);
+  } catch (::NeXus::Exception const &e) {
+    std::cout << "NXRoot: Error loading " << m_filename << "\" in read mode: " << e.what() << "\n";
```

Comment:
Please change this to a `g_log` rather than directly to `std::out`. Even better, delete the print statement altogether.